### PR TITLE
Modal component refactor

### PR DESCRIPTION
**Pull request by AlicjaSzu**
I want to merge this change because...

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

File: src/@next/components/molecules/CardHeader/CardHeader.tsx [hunk 1]
```diff
@@ -24,9 +23,7 @@ export const CardHeader: React.FC<IProps> = ({
   return (
     <S.Header divider={divider}>
       <Text {...getTitleProps(textStyle, titleSize)}>{children}</Text>
-      {closeIcon && (
-        <S.CloseBtn onClick={onHide}>{renderCloseIcon()}</S.CloseBtn>
-      )}
+      {onHide && <S.CloseBtn onClick={onHide}>{renderCloseIcon()}</S.CloseBtn>}
       {customIcon && <S.CloseBtn>{customIcon}</S.CloseBtn>}
```
Comment (jwm0) [inline src/@next/components/molecules/CardHeader/CardHeader.tsx]: As discussed in private: condition and CloseBtn wrapper

File: src/@next/components/organisms/Modal/test.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        <Children />
+      </Modal>
+    );
+  it("exists", () => {
+    const modal = renderModal(DEFAULT_PROPS);
+
+    expect(modal.exists()).toEqual(true);
+  });
+
+  it("render <Overlay /> component with [position, show, hide, target] props", () => {
```
Comment (jwm0) [inline src/@next/components/organisms/Modal/test.tsx]: it render**s**

File: src/checkout/components/AddressPicker/Picker.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import classNames from "classnames";
 import React from "react";
 import ReactSVG from "react-svg";
 
-import { Modal } from "../../../components";
+import { Modal } from "@components/organisms";
```
Comment (jwm0) [inline src/checkout/components/AddressPicker/Picker.tsx]: newline

File: src/@next/components/organisms/Modal/test.tsx (hunk 2 above)
Comment (AlicjaSzu) [reply src/@next/components/organisms/Modal/test.tsx]: it should be 'should render', fixed ;)

File: src/@next/components/molecules/CardHeader/CardHeader.tsx (hunk 1 above)
Comment (AlicjaSzu) [reply src/@next/components/molecules/CardHeader/CardHeader.tsx]: fixed, and Icon replaced with IconButton

**Pull request merged**